build/Wikibase/entrypoint.sh

```diff
@@ -1,59 +1,90 @@
-    echo "$i is required but isn't set. You should pass it to docker. See: https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file";
-    exit 1;
-    fi
-done
+# Exit immediately with message if no /config volume is available
+if [ ! -d "/config" ]; then
+    echo "A volume mapped to /config is required."
+    exit 1
+fi
 
+# Exit immediate on errors or unset variables from here onwards
 set -eu
 
-# Wait for the db to come up
-/wait-for-it.sh "$DB_SERVER" -t 300
-# Sometimes it appears to come up and then go back down meaning MW install fails
-# So wait for a second and double check!
-sleep 1
-/wait-for-it.sh "$DB_SERVER" -t 300
+if [ -e "/config/LocalSettings.php" ]; then
+    cp /config/LocalSettings.php /var/www/html/LocalSettings.php
+    # Always run update (this might be the first run off of a new image version on existing config and data)
+    php /var/www/html/maintenance/run.php update --quick
+else
+    echo "/config/LocalSettings.php not found, running MediaWiki install."
+
+    # Check for required  env vars
+    set +u
+    required_vars=(
+        MW_WG_SERVER
+        DB_USER
+        DB_PASS
+        DB_NAME
+        DB_SERVER
+        MW_ADMIN_PASS
+        MW_WG_LANGUAGE_CODE
+        MW_WG_SITENAME
+        MW_ADMIN_NAME
+        MW_ADMIN_EMAIL
+    )
+    for var in "${required_vars[@]}"; do
+        if [ -z "${!var}" ]; then
+            echo "$var is required but isn't set. You should pass it to Docker. See: https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file"
+            exit 1
+        fi
+    done
+    set -u
 
-# Do the mediawiki install (only if LocalSettings doesn't already exist)
-if [ ! -e "/var/www/html/LocalSettings.php" ]; then
-    php /var/www/html/maintenance/install.php --dbuser "$DB_USER" --dbpass "$DB_PASS" --dbname "$DB_NAME" --dbserver "$DB_SERVER" --lang "$MW_SITE_LANG" --pass "$MW_ADMIN_PASS" "$MW_SITE_NAME" "$MW_ADMIN_NAME"
-    php /var/www/html/maintenance/resetUserEmail.php --no-reset-password "$MW_ADMIN_NAME" "$MW_ADMIN_EMAIL"
+    # Run MediaWiki install script, and update values
+    php /var/www/html/maintenance/run.php install \
+        --server "$MW_WG_SERVER" \
+        --scriptpath "/w" \
+        --dbuser "$DB_USER" \
+        --dbpass "$DB_PASS" \
+        --dbname "$DB_NAME" \
+        --dbserver "$DB_SERVER" \
+        --pass "$MW_ADMIN_PASS" \
+        --lang "$MW_WG_LANGUAGE_CODE" \
+        "$MW_WG_SITENAME" \
+        "$MW_ADMIN_NAME"
 
-    # Copy our LocalSettings into place after install from the template
-    # https://stackoverflow.com/a/24964089/4746236
-    export DOLLAR='$'
-    envsubst < /LocalSettings.php.template > /var/www/html/LocalSettings.php
+    # Include WBS customizations to generated LocalSettings.php
+    {
+        set +u
+        if [ -n "$ELASTICSEARCH_HOST" ]; then
+            echo "\$elasticsearchHost = '$ELASTICSEARCH_HOST';"
```

Comment:
This is instead of using an `envsubst` "template". This is to force `ELASTICSSEARCH_HOST` to be set statically in `LocalSettings.php` after initial install so it operates the same as the rest of the container environment variables. This creates predictable behaviour within the container and makes the `config/LocalSettings.php` generated after initial container run the single source of configuration for the container.

Reply:
could be marked as resolved for now imho